src/librustc_resolve/lib.rs

```diff
@@ -666,7 +670,9 @@ enum RibKind<'a> {
     // methods. Allow references to ty params that impl or trait
     // binds. Disallow any other upvars (including other ty params that are
     // upvars).
-    MethodRibKind,
+    //
+    // The boolean value represents the fact that this method is static or not.
+    MethodRibKind(bool),
```

Comment:
(its good you put the comment here. You *might* consider the alternatives I outlined above, to make the types self-documenting.)